core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/cluster/Cluster.java

```diff
@@ -74,6 +74,9 @@
     @OneToOne
     private Stack stack;
 
+    @Column(columnDefinition = "TEXT")
+    private String variant;
```

Comment:
Wouldn't it be simpler in long terms to use enums

Reply:
Maybe it is not a good idea to store Cluster variants in the core module. We want to move out cluster related stuffs to external modules in the future and remove all cluster-specific related codes from core module like 
```suggestion
        private void prepareClusterManagerVariant(Cluster cluster) {
        if (blueprintService.isAmbariBlueprint(cluster.getBlueprint())) {
            cluster.setVariant("AMBARI");
        } else {
            cluster.setVariant("CLOUDERA_MANAGER");
        }
    }
```

Reply:
should we fix it now or address later?

Reply:
IMHO it is a very big change to move every cluster(-manager) specific codes from core module to separated modules. We should do this, but in later PRs.

Reply:
And if we move the things out, then maybe it is appropriate in this form as it is, because variant will be dynamic and we shouldn't put enums like Ambari, ClouderaManager in the core packages.